local.tf

```diff
@@ -0,0 +1,566 @@
+  ]
+  iam_database_authentication_enabled = true
+
+  database_name   = "wandb_local"
+  master_username = "wandb"
+  master_password = "wandb_root_password"
+
+  vpc_security_group_ids = [aws_security_group.metadata_store.id]
+
+  storage_encrypted = true
+}
+
+resource "aws_rds_cluster_instance" "metadata_store" {
+  identifier           = "wandb-metadata"
+  engine               = "aurora-mysql"
+  cluster_identifier   = aws_rds_cluster.metadata_cluster.id
+  instance_class       = "db.r5.large"
+  db_subnet_group_name = aws_db_subnet_group.metadata_subnets.name
+}
+
+resource "aws_security_group" "metadata_store" {
+  name        = "wandb-metadata-store"
+  description = "Allow inbound traffic from workers to metadata store"
+  vpc_id      = aws_vpc.wandb.id
+
+  tags = {
+    Name = "wandb-metadata-store"
+  }
+}
+
+resource "aws_security_group_rule" "metadata_ingress_eks_workers" {
+  description              = "Allow inbound traffic from EKS workers to metadata store"
+  from_port                = 3306
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.metadata_store.id
+  source_security_group_id = aws_eks_cluster.wandb.vpc_config[0].cluster_security_group_id
+  to_port                  = 3306
+  type                     = "ingress"
+}
+
+##########################################
+# Output files
+##########################################
+
+resource "local_file" "kubeconfig" {
+  filename = "kubeconfig.yaml"
+  content = <<KUBECONFIG
+    apiVersion: v1
+    kind: Config
+    clusters:
+    - cluster:
+        server: ${aws_eks_cluster.wandb.endpoint}
+        certificate-authority-data: ${aws_eks_cluster.wandb.certificate_authority[0].data}
+      name: kubernetes
+    contexts:
+    - context:
+        cluster: kubernetes
+        user: aws
+      name: aws
+    current-context: aws
+    preferences: {}
+    users:
+    - name: aws
+      user:
+        exec:
+          apiVersion: client.authentication.k8s.io/v1alpha1
+          command: aws-iam-authenticator
+          args:
+            - "token"
+            - "-i"
+            - "wandb"
+KUBECONFIG
+}
+
+resource "local_file" "wandb_kube" {
+  filename = "wandb.yaml"
+  content = <<KUBE
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: wandb
+  labels:
+    app: wandb
+spec:
+  strategy:
+    type: RollingUpdate
+  replicas: 1
+  selector:
+    matchLabels:
+      app: wandb
+  template:
+    metadata:
+      labels:
+        app: wandb
+    spec:
+      containers:
+        - name: wandb
+          env:
+            - name: BUCKET
+              value: s3:///${aws_s3_bucket.file_storage.bucket}
+            - name: BUCKET_QUEUE
+              value: sqs://${aws_sqs_queue.file_metadata.name}
+            - name: AWS_REGION
+              value: ${aws_s3_bucket.file_storage.region}
+            - name: MYSQL
+              value: mysql://wandb:wandb_root_password@${aws_rds_cluster_instance.metadata_store.endpoint}/wandb_local
```

Comment:
Would be nice to have this password randomly generated.

Reply:
agree, but the database isn't internet-accessible by default, so the risk is minimal. in the other TF stack, this is parameterized via a variable. there is likely a terraform util to generate something random, but it would be generating the random thing on each `apply`...which would be bad.